### PR TITLE
SageMaker algorithm notebook for computer vision defect detection model

### PR DESCRIPTION
**Pull request by minlu1021**
*Issue #, if available:*

*Description of changes:*

Create SageMaker Algorithm notebook for computer vision defect detection model. 
Next step is to upload model link, once algorithm is published on marketplace.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: computer-vision-defect-detection/Computer Vision Defect Detection Model (Trainable Algorithm).ipynb
Comment (spopov812) [inline computer-vision-defect-detection/Computer Vision Defect Detection Model (Trainable Algorithm).ipynb:1]:
```markdown
1. Note: This notebook contains elements which render correctly in Jupyter interface. Open this notebook from an Amazon SageMaker Notebook Instance or Amazon SageMaker Studio.
1. Ensure that IAM role used has **AmazonSageMakerFullAccess**
1. Some hands-on experience using **Amazon SageMaker**.
1. To use this algorithm successfully, ensure that:
   
   A. Either your IAM role has these three permissions and you have authority to make AWS Marketplace subscriptions in the AWS account used:
   
        a. aws-marketplace:ViewSubscriptions
```
nit: spacing

Comment (spopov812) [inline computer-vision-defect-detection/Computer Vision Defect Detection Model (Trainable Algorithm).ipynb:1]:
```markdown
**Prepare your classification data:**
For this step, we'll follow the data preparation guidelines as outlined in the Amazon Lookout for Vision Developer Guide (https://docs.aws.amazon.com/lookout-for-vision/latest/developer-guide/getting-started.html). And we will use cookie dataset in this guide.

a. Organize your images:
```
Should we add a sample s3 path?

Comment (spopov812) [inline computer-vision-defect-detection/Computer Vision Defect Detection Model (Trainable Algorithm).ipynb:1]:
```markdown
**Prepare your classification data:**
For this step, we'll follow the data preparation guidelines as outlined in the Amazon Lookout for Vision Developer Guide (https://docs.aws.amazon.com/lookout-for-vision/latest/developer-guide/getting-started.html). And we will use cookie dataset in this guide.

a. Organize your images:
Place your normal (non-defective) images in a S3 path named "normal".
Place your anomalous (defective) images in a S3 path named "anomaly".

b. Create a manifest file: The manifest file is a JSON Lines file that lists your images and their classifications. Each line in the file represents one image and contains a JSON object with the following structure:
```
Should we add a sample manifest for jsonl structure?

Comment (spopov812) [inline computer-vision-defect-detection/Computer Vision Defect Detection Model (Trainable Algorithm).ipynb:1]:
```python
                }
            },
            'CompressionType': 'None',
            'RecordWrapperType': 'RecordIO',
            'InputMode': 'Pipe'
        },
    ],
    OutputDataConfig={'S3OutputPath': 's3://'+bucket+'/'+project+'/output'},
```
nit: fstring for consistency

Comment (spopov812) [inline computer-vision-defect-detection/Computer Vision Defect Detection Model (Trainable Algorithm).ipynb:1]:
```python
            'CompressionType': 'None',
            'RecordWrapperType': 'RecordIO',
            'InputMode': 'Pipe'
        },
    ],
    OutputDataConfig={'S3OutputPath': 's3://'+bucket+'/'+project+'/output'},
    ResourceConfig={
        'InstanceType': 'ml.g4dn.2xlarge',
```
Should we document all instance types we support from our training state machine?

Comment (spopov812) [inline computer-vision-defect-detection/Computer Vision Defect Detection Model (Trainable Algorithm).ipynb:1]:
```python
        TrainingJobName=classification_training_job_name
    )
    if training_response['TrainingJobStatus'] == 'InProgress':
        print(".", end='')
    elif training_response['TrainingJobStatus'] == 'Completed':
        print("Completed")
        break
    elif training_response['TrainingJobStatus'] == 'Failed':
```
suggest: adding logging for failure reason

Comment (spopov812) [inline computer-vision-defect-detection/Computer Vision Defect Detection Model (Trainable Algorithm).ipynb:1]:
```markdown
        "type": "groundtruth/image-classification",
        "confidence": 1
    },
    "anomaly-mask-ref": "s3://path-to-image",
    "anomaly-mask-ref-metadata": {
        "internal-color-map": {
            "0": {
                "class-name": "BACKGROUND",
```
unrelated to this PR: how will customers generate manifests with our custom jsonl schema? Is it supported in SMGT?

Comment (minlu1021) [reply computer-vision-defect-detection/Computer Vision Defect Detection Model (Trainable Algorithm).ipynb:1]: Sure, I will add supported list in notebook as well

Comment (minlu1021) [reply computer-vision-defect-detection/Computer Vision Defect Detection Model (Trainable Algorithm).ipynb:1]: Good point, I will add that

Comment (minlu1021) [reply computer-vision-defect-detection/Computer Vision Defect Detection Model (Trainable Algorithm).ipynb:1]: I added the dataset inside the github repo

Comment (minlu1021) [reply computer-vision-defect-detection/Computer Vision Defect Detection Model (Trainable Algorithm).ipynb:1]: They could still use the old way to generate the manifest file, there will be no difference from before.

**Comment by minlu1021** [comment]
Need to publish and merge notebook first, and link to the listing.

**Pull request merged**